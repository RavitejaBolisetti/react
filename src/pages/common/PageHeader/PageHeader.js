import React from 'react';
import { Row, Col, Space, Button, Modal } from 'antd';
import { useNavigate } from 'react-router-dom';
import { ExclamationCircleFilled } from '@ant-design/icons';
import { FaHeart, FaHistory, FaLongArrowAltLeft, FaRegHeart } from 'react-icons/fa';

import { addToolTip } from 'utils/customMenuLink';
import { ROUTING_DASHBOARD } from 'constants/routing';
import styles from './PageHeader.module.css';

const { confirm } = Modal;

export const PageHeader = ({ pageTitle, isFavourite, handleFavouriteClick, visibleChangeHistory = true, handleChangeHistoryClick = undefined, isChangeHistoryVisible = false }) => {
    const navigate = useNavigate();
    const handleBack = () => {
        confirm({
            title: 'Are you sure to leave this page?',
            icon: <ExclamationCircleFilled />,
            content: 'If you leave this page, All unsaved data will be lost',
            okText: 'Yes',
            okType: 'danger',
            cancelText: 'No',
            cancelType: 'danger',
            onOk() {
                navigate(-1) || navigate(ROUTING_DASHBOARD);
            },
            onCancel() {},
        });
    };

    return (
        <>
            <Row gutter={20}>
                <Col xs={16} sm={24} md={12} lg={18} xl={18} xxl={18} className={styles.padRight0}>
                    <Space>
                        <div>
                            <span className={styles.headingGradient}>{pageTitle}</span>
                        </div>
                        <div className={styles.favIconHeading}>{isFavourite ? addToolTip('Mark as unfavourite')(<FaHeart color="#ff3e5b" size={18} onClick={handleFavouriteClick} />) : addToolTip('Mark as favourite')(<FaRegHeart size={18} onClick={handleFavouriteClick} />)}</div>
                    </Space>
                </Col>
                <Col xs={8} sm={24} md={12} lg={6} xl={6} xxl={6} className={styles.padRight0}>
                    <div className={styles.buttonContainer}>
                        {visibleChangeHistory &&
                            (isChangeHistoryVisible ? (
                                <Button type="primary" onClick={handleChangeHistoryClick}>
                                    <FaHistory className={styles.buttonIcon} />
                                    Change History
                                </Button>
                            ) : (
                                <Button danger onClick={handleChangeHistoryClick}>
                                    <FaHistory className={styles.buttonIcon} />
                                    Change History
                                </Button>
                            ))}

                        <Button danger onClick={handleBack} className="btn btn-outline srchbtn mr0 boxShdwNon">
                            <FaLongArrowAltLeft className={styles.buttonIcon} />
                            Exit
                        </Button>


                       

                    </div>
                </Col>
            </Row>

            <Row gutter={20}>
                <Col xs={24} sm={24} md={12} lg={24} xl={24} xxl={24}>
                    <div className={styles.pageHeaderNameSection}></div>
                </Col>
            </Row>
        </>
    );
};
