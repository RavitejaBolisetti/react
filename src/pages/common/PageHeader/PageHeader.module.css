.buttonContainer :global .ant-btn-default.ant-btn-dangerous:not(:disabled):hover {
    background-color: #ff3e5b;
    color: #ffffff !important;
}

:global .ant-btn {
    height: 34px;
    padding: 6px 12px;
    border-radius: 4px;
}

.padRight0 {
    padding-right: 0px !important;
}

.pageHeader .headingGradient {
    padding-bottom: 10px;
    border-radius: 5px;
    border: 0px;
    border-bottom: 4px solid;
    border-image: linear-gradient(to right, #efa3a3, #cac9ee);
    border-image-slice: 1;
    text-decoration: none;
    margin-bottom: 28px;
    color: #0b0b0c;
    font-size: 20px;
}

.pageHeader .favIconHeading {
    padding-left: 5px;
    margin-top: 5px;
    line-height: 4px;
    padding-top: 3px;
    cursor: pointer;
    color: #ff3e5b;
}

.pageHeaderNameSection {
    border-bottom: 1px solid #dedede;
    margin: 0 0 16px 0;
}

.pageHeader .buttonContainer {
    text-align: right;
}

.pageHeader .buttonContainer button {
    margin-left: 14px;
}

.pageHeader .buttonContainer .buttonIcon {
    margin-right: 6px;
    padding-top: 2px;
}

/* Modal Content */

:global .ant-modal-confirm .ant-modal-confirm-body .ant-modal-confirm-title {
    display: contents;
}

:global .ant-modal-confirm .ant-modal-confirm-btns {
    text-align: center;
    margin-top: 0;
}

:global .ant-modal-confirm .ant-modal-confirm-body .ant-modal-confirm-title+.ant-modal-confirm-content {
    max-width: 100%;
    padding: 10px;
    text-align: justify;
    border-top: 1px solid #DEDEDE;
}

:global .ant-modal-confirm .ant-modal-confirm-btns button:last-child {
    background-color: #ff3e5b;
    color: #ffffff !important;
}

:global .ant-modal-confirm .ant-modal-confirm-btns button {
    text-align: center;
    width: 100px;
    margin-top: 15px;
    background-color: #ffffff;
    border: 1px solid #ff3e5b;
    color: #ff3e5b !important;
}

:global .ant-modal-confirm .ant-modal-confirm-btns button:hover {
    background-color: #ff3e5b;
    border: 1px solid #ff3e5b;
    color: #ffffff !important;
}

.modalIconAlert {
    color: #FF7A00;
    padding-right: 8px;
}

@media (max-width: 768px) {

    .pageHeader .headingGradient {
        padding-bottom: 5px;
        /* border-bottom: 0px; */
        font-size: 14px;
        font-weight: 600;
    }

    .pageHeaderNameSection {
        border-bottom: 0;
        padding: 0px;
        margin: 0 0 10px 0;
        /* width: 68vw; */
    }

    .pageHeader .buttonContainer {
        margin-top: 6px;
    }

    .pageHeader .buttonContainer button {
        margin-left: 5px;
    }

    .pageHeader .buttonContainer :global .ant-btn {
        font-size: 12px;
        height: 24px;
        padding: 2px 6px;
    }

    .pageHeader .buttonContainer .buttonIcon {
        margin-right: 2px;
    }

    .buttonContainer button {
        margin-left: 0px;
    }

}