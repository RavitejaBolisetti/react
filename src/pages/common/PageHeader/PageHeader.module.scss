@import 'assets/sass/app.module.scss';

.pageHeader {
    height: 37px;

    & :global {
        .ant-btn {
            height: 34px;
            @include space(padding, top, $sp4);
            @include space(padding, right, $sp12);
            @include space(padding, bottom, $sp4);
            @include space(padding, left, $sp12);
            @include borderRedius(4px, 1px, solid, inherit);
        }
    }

    .headingGradient {
        //padding-bottom: 12px;
        @include space(padding, bottom, $sp13);
        // @include space(margin, bottom, $sp28);
        @include borderRedius(4px, 0px, solid, inherit);
        border-bottom: 4px solid;
        @include bordergradient($DarkPink, $LightPink);
        border-image-slice: 1;
        @include font-style(16, 24, 700, $primaryBlack, $fontfamily, left);
        text-decoration: none;

        @include breakpoint(md) {
            padding-bottom: 5px;
            @include space(padding, bottom, $sp5);
            @include font-style(14, 24, 600, $primaryBlack, $fontfamily, left);
        }

        .favIconHeading {
            @include space(padding, top, $sp3);
            @include space(margin, top, $sp5);
            @include space(margin, left, $sp14);
            line-height: 4px;
            cursor: pointer;
            color: $primaryRed;
        }
    }

    .buttonContainer {
        text-align: right;

        @include breakpoint(md) {
            @include space(margin, top, $sp6);
        }

        & :global {
            .ant-btn {
                @include breakpoint(md) {
                    //font-size: 12px;
                    @include font-style(12, inherit, normal, normal, $fontfamily, right);
                    height: 24px;
                    @include space(padding, top, $sp2);
                    @include space(padding, right, $sp6);
                    @include space(padding, bottom, $sp2);
                    @include space(padding, left, $sp6);
                }
            }

            .ant-btn-default {
                &.ant-btn-dangerous {
                    &:not(:disabled):hover {
                        background-color: $primaryRed;
                        color: $primaryWhite !important;
                    }
                }
            }
        }

        button {
            margin-left: 14px;

            @include breakpoint(md) {
                @include space(margin, left, $sp5);
                // margin-left: 0px;
            }
        }

        .buttonIcon {
            @include space(padding, top, $sp2);
            @include space(margin, right, $sp6);

            @include breakpoint(md) {
                @include space(margin, right, $sp2);
            }
        }
    }
}

.pageHeaderNameSection {
    border-bottom: 1px solid $neutralGrey1;
    @include space(margin, top, $sp1);
    @include space(margin, bottom, $sp20);

    @include breakpoint(md) {
        border-bottom: 0;
        @include space(padding, all, $sp0);
        @include space(margin, top, $sp0);
        @include space(margin, right, $sp0);
        @include space(margin, bottom, $sp10);
        @include space(margin, left, $sp0);
    }
}

/* Modal Content */
:global {
    .ant-modal-confirm {
        .ant-modal-confirm-body {
            .ant-modal-confirm-title {
                display: contents;
            }

            .ant-modal-confirm-title + .ant-modal-confirm-content {
                max-width: 100%;
                padding: 10px;
                text-align: justify;
                border-top: 1px solid $neutralGrey1;
            }
        }

        .ant-modal-confirm-btns {
            text-align: center;
            margin-top: 0;

            button {
                &:last-child {
                    background-color: $primaryRed;
                    color: $primaryWhite !important;
                }

                text-align: center;
                width: 100px;
                margin-top: 15px;
                background-color: $primaryWhite;
                border: 1px solid $primaryRed;
                color: $primaryRed !important;

                &:hover {
                    background-color: $primaryRed;
                    border: 1px solid $primaryRed !important;
                    color: $primaryWhite !important;
                }
            }
        }
    }
}
