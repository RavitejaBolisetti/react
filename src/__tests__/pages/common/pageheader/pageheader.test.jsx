import '@testing-library/jest-dom/extend-expect';
import customRender from '@utils/test-utils';
import PageHeader from '@pages/common/PageHeader/PageHeader';
import { screen, fireEvent} from "@testing-library/react";

const mockFetchList = jest.fn();
const mockMarkFavourite = jest.fn();
const mockListShowLoading = jest.fn();
const mockShowGlobalNotification = jest.fn();

describe('PageHeader Component', () => {
  it('should render PageHeader component', async () => {
      const { container } = customRender(<PageHeader isFavourite={false} canMarkFavourite={true} mockFetchList={mockFetchList} mockMarkFavourite={mockMarkFavourite} mockShowGlobalNotification={mockShowGlobalNotification} mockListShowLoading={mockListShowLoading} />);
      expect(container).toMatchSnapshot();
      screen.debug();
  });

  it('renders page title correctly', () => {
    customRender(<PageHeader pageTitle="Test Page" />);
    const pageTitleElement = screen.getByText('Test Page');
    expect(pageTitleElement).toBeInTheDocument();
  });

 it('displays "Mark as favorite" icon when canMarkFavourite is true', () => {
    customRender(<PageHeader pageTitle="Mark as favourite" canMarkFavourite={true} onError={jest.fn()} handleFavouriteClick={jest.fn()} isFavourite={false}/>);
    const markAsFavoriteIcon = screen.getByTestId('addfav');
    expect(markAsFavoriteIcon).toBeInTheDocument();
 });

 it('displays "Mark as favorite" icon when canMarkFavourite is true', () => {
    customRender(<PageHeader pageTitle="Mark as favourite" canMarkFavourite={true} onError={jest.fn()} handleFavouriteClick={jest.fn()} isFavourite={false}/>);
    const markAsFavoriteIcon = screen.getByTestId('addfav');
    fireEvent.click(markAsFavoriteIcon);
  });

  it('calls handleFavouriteClick when "Mark as favorite" icon is clicked', () => {
    customRender(<PageHeader pageTitle="Test Page" canMarkFavourite={true}  onError={jest.fn()} handleFavouriteClick={jest.fn()} isFavourite={true} />);
    const markAsFavoriteIcon = screen.getByTestId('addfav');
    fireEvent.click(markAsFavoriteIcon);
    expect(mockMarkFavourite).toHaveBeenCalledTimes(1);
  });
});