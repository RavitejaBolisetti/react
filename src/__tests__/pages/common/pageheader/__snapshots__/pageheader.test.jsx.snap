// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Component should render PageHeader component 1`] = `
<div>
  <div
    class="ant-row pageHeader css-dev-only-do-not-override-14wwjjs"
    style="margin-left: -10px; margin-right: -10px;"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 ant-col-xxl-24 css-dev-only-do-not-override-14wwjjs"
      style="padding-left: 10px; padding-right: 10px;"
    >
      <div
        class="ant-space css-dev-only-do-not-override-14wwjjs ant-space-horizontal ant-space-align-center"
      >
        <div
          class="ant-space-item"
        >
          <div>
            <span
              class="headingGradient"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-row css-dev-only-do-not-override-14wwjjs"
    style="margin-left: -10px; margin-right: -10px;"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-sm-24 ant-col-md-24 ant-col-lg-24 ant-col-xl-24 ant-col-xxl-24 css-dev-only-do-not-override-14wwjjs"
      style="padding-left: 10px; padding-right: 10px;"
    >
      <div
        class="pageHeaderNameSection"
      />
    </div>
  </div>
  ;
</div>
`;
